assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * AdSense Setup UseSnippet Switch Component.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { MODULES_ADSENSE } from '../../../datastore/constants';
+import { parseAccountID } from '../../../util/parsing';
+import { UseSnippetSwitch } from '../../common';
+const { useSelect, useDispatch } = Data;
+
+export default function SetupUseSnippetSwitch() {
+	const originalUseSnippet = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getOriginalUseSnippet()
+	);
+	const existingTag = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getExistingTag()
+	);
+	const clientID = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getClientID()
+	);
+
+	const { setUseSnippet, saveSettings } = useDispatch( MODULES_ADSENSE );
+
+	const hasExistingTag = Boolean( existingTag );
+	useEffect( () => {
+		( async function () {
+			if ( hasExistingTag ) {
+				setUseSnippet( false );
+				await saveSettings();
+			}
+		} )();
```

Comment:
Thanks, @hussain-t, but I think there is no need to wait till the saveSettings function call finishes. Let's remove await keyword and wrapping async function.

Reply:
You're right 👍